modules/xmpp/XmppConnection.js

```diff
@@ -432,8 +432,33 @@ export default class XmppConnection extends Listenable {
         const { shard, websocketKeepAliveUrl } = this._options;
         const url = websocketKeepAliveUrl ? websocketKeepAliveUrl
             : this.service.replace('wss://', 'https://').replace('ws://', 'http://');
-
+        
+        // #bloomberg #shard @rpang27 use prodody url to get shard
+        const getShardUrl = this.service.replace('wss://', 'https://').replace('ws://', 'http://').replace('xmpp-websocket', 'shard');
+        // #end
+        
         return fetch(url)
+            // #bloomberg #shard @rpang27 do not check shard from keep alive response
+            // .then(response => {
+            //     // skips header checking if there is no info in options
+            //     if (!shard) {
+            //         return;
+            //     }
+                
+            //     const responseShard = response.headers.get('x-jitsi-shard');
+                
+            //     if (responseShard !== shard) {
+            //         logger.error(
+            //             `Detected that shard changed from ${shard} to ${responseShard}`);
+            //             this.eventEmitter.emit(XmppConnection.Events.CONN_SHARD_CHANGED);
+            //         }
+            // })
+            // #end
+            .catch(error => {
+                logger.error(`Websocket Keep alive failed for url: ${url}`, { error });
+            })
+            // #bloomberg #shard @rpang27 check shard from prosody response
+            .then(() => fetch(getShardUrl))
             .then(response => {
 
                 // skips header checking if there is no info in options
```

Comment:
You can skip the `fetch` request for shard if `shard` is not set in options.

Reply:
done